### ISSUE TITLE
Can't run differential delete

### ISSUE THREAD

**Issue by benblamey** [MathieuLoutre/grunt-aws-s3#99]
I'm currently using the "upload" action; without any problem. When I try to perform a differential delete, I get the following error:
"Fatal error: Differential delete needs a "cwd"

This is my grunt.js
```javascript
module.exports = function(grunt) {
  grunt.initConfig({
    pkg: grunt.file.readJSON('package.json'),

    aws_s3: {
      options: {
        region: 'eu-west-1',
        uploadConcurrency: 5,
        downloadConcurrency: 5,
      },

      dist: {
        options: {
          bucket: 'some-bucket-name',
        },
        files: [
          {
            expand: true,
            cwd: 'pages/',
            src: ['**'],
            dest: '/',
            differential: true,
            action: 'delete'
          }
        ],
      },
    },
  });

  grunt.loadNpmTasks('grunt-aws-s3');
  grunt.registerTask('deploy', [
    'aws_s3:dist',
  ]);
};
```

**Comment by benblamey** [comment]
I briefly attempted to debug the issue, and noticed the following in aws_s3.js
![gruntjs - cwd issue](https://cloud.githubusercontent.com/assets/782539/12453713/cba37dd2-bf94-11e5-87dd-ab745287f816.png)

**Comment by MathieuLoutre** [comment]
Sorry! Only just seen this. Thanks for having a deeper look as well. Maybe this line should read `filePair.differential && !filePair.orig.cwd`? Can you try it that way. If it does work for you, I'm happy to take a PR or I can make the change if you prefer. Let me know!

**Comment by benblamey** [comment]
I tried the change as suggested - and I get a different error:

````
Running "aws_s3:dist" (aws_s3) task
Deleting the content of https://s3-eu-west-1.amazonaws.com/xxxxxx/
Fatal error: Path must be a string. Received undefined
````

**Comment by belghitial** [comment]
Hello i see in README file that (delete: will only delete the files which don't exist locally) but no example how it work, can anyone tell me how ? 

Thanks.

**Comment by belghitial** [comment]
j'ai trouvé il fallait que je rajoute un / a dest et mettre differential: true et cwd dans les options.

Merci.

**Comment by trioni** [comment]
Any updates about this?

**Comment by MathieuLoutre** [comment]
@trioni do you have the same error as OP or something different?

**Comment by trioni** [comment]
@MathieuLoutre yes, this error `Fatal error: Differential delete needs a "cwd"`.

Currently using npm-scripts and the aws cli to achieve the same thing:

```
AWS_PROFILE=myprofile aws s3 sync ./public/build-$npm_package_version s3://mybucket/public --delete --region some-region
```

**Comment by parsingphase** [comment]
Having tested a few ways it looks like slashes in dirs aren't to blame, but using expand:true where action is delete causes the problem.

Given that no src is needed where action = delete, expand presumably doesn't do anything anyway.

ie, this works unless I uncomment the expand line:

               {
                        action: 'delete', // remove any files not longer present locally
                        // expand: true,
                        // src: '**/*.png',
                        dest: 'domain_com/d/static-testing', /* s3 destination */
                        differential: true,
                        cwd: staticRootDir // does not end with '/'
                    }
